src/mysql_protocol.erl

```diff
@@ -28,7 +28,7 @@
 -module(mysql_protocol).
 
 -export([handshake/8, change_user/8, quit/2, ping/2,
-         query/6, fetch_query_response/5, prepare/3, unprepare/3,
+         query/6, fetch_query_response/5, prepare/3, unprepare/3, prepare/4,
```

Comment:
Please delete prepare/3 and change all the calls to use prepare/4 instead.

Reply:
ah! This would break existing applications which depend on prepare/3. I am not sure if that would be a great idea.

Reply:
In `mysql` we must keep prepare/3 in of course, but `mysql_protocol` is an internal module. We have removed arity variants here before, not long ago. That's why there is only query/6 for query for example.